### PR TITLE
fix the build

### PR DESCRIPTION
**Pull request by popey**
fixing the build of corsix-th. bumped to core22 and stripped down the libraries too.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -13,16 +13,17 @@ description:
...
 confinement: strict
 
 apps:
   corsixth:
-    command: desktop-launch $SNAP/bin/corsix-th $SNAP/share/corsix-th/CorsixTH.lua
+    extensions: [ gnome ]
+    command: bin/launch
     environment:
-      LD_LIBRARY_PATH: '$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/lua/5.3'
+      #LD_LIBRARY_PATH: '$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/lua/5.3'
```
Comment (sergiusens) [inline snap/snapcraft.yaml]:
```suggestion
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -13,16 +13,17 @@ description:
...
+base: core22
 
 grade: stable
 confinement: strict
 
 apps:
   corsixth:
-    command: desktop-launch $SNAP/bin/corsix-th $SNAP/share/corsix-th/CorsixTH.lua
+    extensions: [ gnome ]
+    command: bin/launch
```
Comment (sergiusens) [inline snap/snapcraft.yaml:24]: Have not tested, but this should work, removing the need for `snap/local/bin/launch`
```suggestion
    command: usr/local/bin/corsix-th --interpreter=$SNAP/usr/local/share/corsix-th/CorsixTH.lua
```

**Review by sergiusens — APPROVED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (popey) [reply snap/snapcraft.yaml:24]: I tried that and snapcraft complained about erroneous characters in the command line, so I moved it out to a launch script.

**Comment by sergiusens** [comment]
Ah, the = sign in the command, which can probably be removed too

Em sáb., 7 de out. de 2023 11:50, Alan Pope ***@***.***>
escreveu:

**Comment by popey** [comment]
Nope. tried that. If you don't specify the = it fails.

```
$ snap run corsixth
CorsixTH cannot find CorsixTH.lua. If you want use a custom location, specify it by --interpreter=FILE

```

**Review by merlijn-sebrechts — APPROVED** [review]
LGTM! I propose merging this to get it working again, and taking a stab at the cosmetic issue later

**Comment by popey** [comment]
It's failing to build now, looking for libmfx1...

https://launchpadlibrarian.net/690973096/buildlog_snap_ubuntu_jammy_arm64_69683c0c6722af9883200c5ffd5f2fa7_BUILDING.txt.gz

but it built fine locally...

```
Initializing parts lifecycle
Executing parts lifecycle...
Executing parts lifecycle: pull corsixth
Executing action
Stage package not found in part 'corsixth': libmfx1.
Full execution log: '/root/.local/state/snapcraft/log/snapcraft-20231008-184145.543169.log'
Build failed
```

**Pull request merged**